fiftyone/core/collections.py

```diff
@@ -3898,6 +3897,93 @@ def list_evaluations(self, type=None, method=None, **kwargs):
             self, type=type, method=method, **kwargs
         )
 
+    def map_samples(
+        self,
+        map_fcn: Callable[[Any], Any],
+        *_,
+        workers: Optional[int] = None,
+        batch_method: str = "id",
+        batch_size: Optional[int] = None,
+        progress: Optional[Union[bool, Literal["worker"]]] = None,
+        parallelize_method: str = "process",
+        skip_failures: bool = False,
+        save: bool = False,
+    ):
+        """
+        Applies `map_fcn` to each sample using the specified backend strategy
+        and returns an iterator.
+
+        Args:
+            map_fcn: Function to apply to each sample.
+            workers (None): Number of workers.
+            batch_method: Explicit method for batching samples. Supported
+              methods are 'id' and 'slice'
```

Comment:
The reader will have no idea what these methods are.

Reply:
Hmm we can add a couple of sentences to explain them here but it might be too long. Should we add a link to documentation instead?